### PR TITLE
Dockerfiles

### PR DESCRIPTION
**Pull request by Octopus773**

### CODE REVIEW

File: images/c/Dockerfile.standalone [hunk 1]
```diff
@@ -1,9 +1,5 @@
 FROM whanos-c:latest
```
Comment (AnonymusRaccoon) [inline images/c/Dockerfile.standalone:1]: A two step dockerfile should be used for this. We don't need gcc/make or source files on the standalone.

It should probably be something like this:

```Dockerfile
FROM whanos-c:latest as builder
COPY . /app
WORKDIR /app
RUN make

FROM alpine
SHELL ["/usr/bin/bash", "-c"]
CMD ["./compiled-app"]
```

PS: I did not test the dockerfile.

File: images/java/Dockerfile.standalone [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM whanos-java:latest
```
Comment (AnonymusRaccoon) [inline images/java/Dockerfile.standalone:1]: Same as the C, this should be a builder and the after the build you should do a FROM java without the sdk, maven and all of that. With just the jvm.

File: images/c/Dockerfile.standalone (hunk 1 above)
Comment (Octopus773) [reply images/c/Dockerfile.standalone:1]: alpine doesn't have bash

Comment (AnonymusRaccoon) [reply images/c/Dockerfile.standalone:1]: FROM bash:latest then

Comment (AnonymusRaccoon) [reply images/c/Dockerfile.standalone:1]: It will be installed to /usr/local/bin/bash if you use this image.

Comment (Octopus773) [reply images/c/Dockerfile.standalone:1]: i will keep this in mind

**Comment by Octopus773** [comment]
Update from Hugo perez all Dockerfiles **must** derive from official docker images (including standalone files)

**Comment by Octopus773** [comment]
`ONBUILD` docker instruction doesn't seem to be usable with multi-stage builds

**Review by AnonymusRaccoon — APPROVED** [review]

File: images/c/Dockerfile.base [hunk 3]
```diff
@@ -1 +1,25 @@
...
+
+# using bash shell
+SHELL ["/bin/bash", "-c"]
+# copying all current directory files in the docker
+ONBUILD COPY . /app
+# going to /app
+ONBUILD WORKDIR /app
+# compiling the binary
+# remove the Makefile + the sources of the binary
+ONBUILD RUN make && rm -rf Makefile app
```
[the remark this answers is not in the document]
Comment (Octopus773) [reply images/c/Dockerfile.base:11]: Later I'm copying the /app folder, and I'm expecting it to only have the binary

**Review by AnonymusRaccoon — APPROVED** [review]

**Pull request merged**